applications/conceptual-model-editor/src/app/diagram/dialog/select-classes-dialog.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useState } from "react";
+import { useModelGraphContext } from "../context/model-context";
+import { isSemanticModelClass } from "@dataspecer/core-v2/semantic-model/concepts";
+import { useBaseDialog } from "../components/base-dialog";
+import { useConfigurationContext } from "../context/configuration-context";
+import { CreateButton } from "../components/dialog/buttons/create-button";
+import { CancelButton } from "../components/dialog/buttons/cancel-button";
+import { t } from "../application/";
+import { isSemanticModelClassUsage } from "@dataspecer/core-v2/semantic-model/usage/concepts";
+import { type EntityDetailSupportedType, EntityProxy } from "../util/detail-utils";
+import { tailwindColorToHex } from "~/app/utils/color-utils";
+
+const getDefaultColor = () => {
+    return "#069420";
+};
+
+export const useSelectClassesDialog = () => {
+    const { isOpen, open, close, BaseDialog } = useBaseDialog();
+    const [onSelectConfirmCallback, setOnSelectConfirmCallback] = useState<((newEntityID: string[]) => void) | undefined>(undefined);
+
+    const localOpen = (onSelectConfirmCallback?: (newEntityID: string[]) => void) => {
+        setOnSelectConfirmCallback(onSelectConfirmCallback);
+        open();
+    };
+
+    // TODO: Maybe could be optimized later, when I get to know React better
+    function createChangedRecord<T>(record: Record<string, T>, id: string, newValue: T): Record<string, T> {
+        const newRecord = {...record};
+        newRecord[id] = newValue;
+        return newRecord;
```

Comment:
I belive this could be simplified ... to almost a one liner 
```js
return {...record, [id]: newValue}.
```
as such it may even be inlined into the code. That is why I do not see too much of a value having this function.

Reply:
Much better. That being said I rewrote it, so we don't use map as values for the combobox. Because on every check the component is rerendered and for big diagram it takes way too long. We are no longer using state for the map, not sure if it is good idea. Right now it works since there are no rerenders or other changing states.